Feature: Sim mode at runtime (#5721)
cylc/cylc-flow: changes.d/5721.feat.md

> Allow users to broadcast simulation mode settings to tasks.

Reviewer: I don't think that's correct?? Run mode applies to the scheduler, not tasks

> Allow users to broadcast simulation mode settings to tasks.

Author: In the past this has been true, but the point of this work is to make Cylc ready for Skip mode, and that requires the run mode to be applied on a task-by-task basis.

I think we should probably add the word "settings" - because these definately are changed on a task by task basis.

> Allow task's run mode settings to be changed dynamically at runtime using `cylc broadcast`.

Reviewer: > ...the point of this work is to make Cylc ready for Skip mode, and that requires the run mode to be applied on a task-by-task basis. Having said which, simulation mode will continue to be a whole workflow setting, at least by default, so you're probably right.
> 
> I think we should probably add the word "settings" - because these definately are changed on a task by task basis.

_https://github.com/cylc/cylc-flow/pull/5721#discussion_r1510912272_

But this PR only enables changing simulation mode settings, right?
```suggestion
Allow tasks' simulation mode settings to be changed dynamically at runtime using `cylc broadcast`.
```

Reviewer 2: Yes, the next, hopefully smaller step, is to allow the run mode to be broadcasted at runtime.

> Allow tasks' simulation mode settings to be changed dynamically at runtime using `cylc broadcast`.

Reviewer 3: ```suggestion
Allow task simulation mode settings to be changed dynamically using `cylc broadcast`.
```

> Allow task's run mode (i.e. "live", "simulation" or "dummy") to be changed dynamically at runtime using `cylc broadcast`.

Author: ```suggestion
Allow task's run mode settings to be changed dynamically at runtime using `cylc broadcast`.
```

> Allow task's run mode (i.e. "live", "simulation" or "dummy") to be changed dynamically at runtime using `cylc broadcast`.

Reviewer: https://github.com/cylc/cylc-flow/pull/5721#discussion_r1507499124

> I don't think that's correct?? Run mode applies to the scheduler, not tasks
